src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -306,10 +306,12 @@ pub enum Connector<T: AstInfo> {
         publication: String,
         /// The namespace the synced table belongs to
         namespace: String,
-        /// The name of the table to sync
-        table: String,
-        /// The expected column schema of the synced table
-        columns: Vec<ColumnDef<T>>,
+        /// The table to sync, if only a single table
+        /// Will be set for `CREATE SOURCE`, but not `CREATE SOURCES`
+        table: Option<PgTable<T>>,
+        /// The tables to sync, if multiple tables
+        /// Will be set for `CREATE SOURCES`, but not `CREATE SOURCE`
+        tables: Option<Vec<PgTable<T>>>,
```

Comment:
Actually, thinking about it more, despite the duplication I think it will be safer if you create an entirely new fork of the AST for the multiple sources case. 

```rust
enum Connector<T> {
    // ...
    Postgres {
        conn: String,
        publication: String,
        table: String,
        columns: Vec<ColumnDef<T>>,
    }
}

enum MultiConnector {
    Postgres {
        conn: String,
        publication: String,
        tables: Vec<PgTable<T>>,
    }
}

pub struct PgTable<T: AstInfo> {
     pub name: UnresolvedObjectName,
     pub alias: Option<T::ObjectName>,
     pub columns: Vec<ColumnDef<T>>,
 }
```

That way in the planner you won't have to be switching on `table` vs `tables`. It also makes me a little nervous when valid ASTs don't serialize to valid SQL; e.g., with the current design, you can construct an AST that renders to "CREATE SOURCE ... TABLES (...)" which is not actually valid SQL.